### ISSUE TITLE
Lovelace entity button double_tap_action causing undesired behavior

### ISSUE THREAD

**Issue by Trumpiter** [home-assistant/home-assistant-polymer#4419]
**Checklist:**

- [x] I updated to the latest version available
- [x] I cleared the cache of my browser

**Home Assistant release with the issue: 0.103.5

**Last working Home Assistant release (if known): unknown

**UI (States or Lovelace UI?): Lovelace UI

**Browser and Operating System: iOS 13.3 (issue occurs in Chrome, Safari, and the companion app)

**Description of problem:
I am attempting to use both tap_action to toggle a single light and double_tap_action to toggle all lights with an entity button in Lovelace. Currently, when both options are specified, they work perfectly on my laptop (Chrome on MacOS 10.14.6), but a single tap on my phone will produce the double tap action.

It looks like it might be related to this issue https://github.com/home-assistant/home-assistant-iOS/issues/431 and the subsequent fix https://github.com/home-assistant/home-assistant-polymer/pull/3523

Further testing: I changed the behavior of two separate entity buttons to further explore the behavior. One I set tap_action to toggle and double_tap_action to none and the second I set tap_action to none and double_tap_action to toggle. On MacOS, the behavior is as expected--the first button toggles with a single tap (and with a double tap) and the second button only toggles with a double tap. On iOS, both buttons toggle with single taps (and with double taps). 

TLDR: On ~~iOS~~ some devices, Lovelace UI does not differentiate between a single and a double tap, and when both actions are specified it defaults to double_tap_action.

**Expected behaviour: One tap -> tap_action. Double tap -> double_tap_action (or tap_action if no double_tap_action is specified)

**Relevant config:
```
double_tap_action:
  action: call-service
  service: light.toggle
  service_data:
    entity_id: group.all_lights
entity: light.bed
hold_action:
  action: more-info
tap_action:
  action: toggle
type: entity-button
```

**Steps to reproduce this problem:**

**Javascript errors shown in the web inspector (if applicable):**

**Additional information:**

**Comment by rvoosterhout** [comment]
I can confirm I also have this problem on my iPads running ios 13.3. My Iphones are fine. Clicking an entity button causes a double tap instead of a toggle.

**Comment by PiotrMachowski** [comment]
I have the same issue on Android 9 devices (Samsung Galaxy S8, Nokia 8, Huawei MediaPad M5) in Google Chrome, Samsung Internet and WallPanel apps.

**Comment by Trumpiter** [comment]
I just checked with my old phone, and this issue occurs with iOS 12.4 on Safari as well.

I edited the title and initial post to reflect that this issue is not restricted to iOS devices.

**Comment by iantrich** [comment]
@bramkragten, @RomRider looks to have a solution for this in the [button-card](https://github.com/custom-cards/button-card/blob/235d025a50439e11ba8c4d4da6090300c78c2520/src/long-press.ts#L184-L187)

**Comment by bramkragten** [comment]
> @bramkragten, @RomRider looks to have a solution for this in the [button-card](https://github.com/custom-cards/button-card/blob/235d025a50439e11ba8c4d4da6090300c78c2520/src/long-press.ts#L184-L187)

We should update our check, not sure if that is the right check...
https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/panels/lovelace/common/directives/action-handler-directive.ts#L167

**Comment by RomRider** [comment]
The core implementation used to work but not anymore with the latest ipadOS @bramkragten.
My solution seems to be work at least with IPadOS

**Comment by bramkragten** [comment]
Yeah, I expected it would break with IPadOS, but the users above mention iOS 12.4... And does your check still works for iPhone's on iOS 13?

**Comment by PiotrMachowski** [comment]
> @bramkragten, @RomRider looks to have a solution for this in the [button-card](https://github.com/custom-cards/button-card/blob/235d025a50439e11ba8c4d4da6090300c78c2520/src/long-press.ts#L184-L187)

I confirm that button-card is working correctly on all my devices (Samsung Galaxy S8, Nokia 8, Huawei MediaPad M5, all running Android 9. Browsers: Google Chrome, Samsung Internet and WallPanel app)

**Comment by RomRider** [comment]
I updated button card again with a fix for older iPhones.
https://github.com/custom-cards/button-card/blob/6945bc007e2ef08cf237c467f2b196dcb7aaa1ca/src/long-press.ts#L184-L187

**Comment by SamLewis0602** [comment]
I am seeing this same behavior using button-card on iOS 13 with the Home Assistant app. 

If I have `double_tap_action` and `tap_action` set on a button, the `double_tap_action` is executed on both single and double taps.
If I have `hold_action` and `tap_action` set on a button, single tap events seem to trigger twice. It will turn lights on and immediately off again on a single tap.

Both of the above scenarios work fine in Chrome on MacOS.

**Comment by RomRider** [comment]
@SamLewis0602 make sure to clear the cache and update to the latest button-card version. It is fixed there.

**Comment by Trumpiter** [comment]
Thanks guys, button-card fixed the problem I was having. I'm still a bit miffed that the native button had issues, but now you've gotten me into custom cards for lovelace!

**Comment by SamLewis0602** [comment]
@RomRider I've confirmed I'm running the latest button-card version via HACS (3.1.1) and pressed "Reset frontend cache" in the App Configuration section in the iOS app, but I'm still seeing this issue. Any ideas?

**Comment by iantrich** [comment]
That would be better discussed in the repo

**Comment by SamLewis0602** [comment]
Agreed, this was the only conversation I could find about the issue. Moved this over to the thread above.

**Comment by ezaton** [comment]
I have to ask to reopen this ticket. It appears to translate any single click to double-click on Andriod devices as well. It happened to me both on the official application and Chrome for Android. Android version is 10.
I am using HassOS, with the production homeassistant docker, and I do not know how to "reload the card".
When double-click action is defined, Android uses it instead of a single-click, when single-clicked.

This is not button-card or custom-card specific. It happens with any regular card allowing double-click action.

HassOS 3.8
Supervisor 195
HomeAssistant 0.104.3

Thanks

**Comment by iantrich** [comment]
This isn't closed?

**Comment by Trumpiter** [comment]
@ezaton Are you using entity-button or button-card? I wasn't able to get the double tap action to work using entity-button but using the custom button-card fixed the issue for me.

**Comment by ezaton** [comment]
I did not use the custom button-card yet. The problem is in the entity-button. Workarounds are fun and great, and I will use them (working on that now), however, the problem is also in the picture-elements card as well, and many others. It means that the workaround, while nice, and would solve some things, is not a *solution*. Therefore, I do not think that this bug is closed. Don't you agree? Either there is a fix in the product (and then the bug is closed, and my task is to update the relevant component or wait for hass.io to update to the next one), or there is a fix elsewhere, as a workaround, and then the bug is still open. That is my view, and that is why I asked to reopen this bug.
Thanks

**Comment by bramkragten** [comment]
But it is open? I'm planning to have this fixed in 0.105

**Comment by ezaton** [comment]
Currently, with version 0.104.3 the bug exists. So, I guess you can say it's open.

Thanks!

**Comment by iantrich** [comment]
@ezaton this has never been closed. I believe you are confused

**Comment by ezaton** [comment]
You are right. It seems that the redirection to the button-card ticket, got me confused. This is an open ticket, and I apologise. Thanks

**Comment by bramkragten** [comment]
@RomRider do I understand correctly that you don't support double clicks on the iPad?
Since the code only checks from the mouseEvent if it is a double click and you don't catch mouseEvents on iPhones and iPads? Or am I missing something?

**Comment by ezaton** [comment]
The recent update to version 0.105.2 has solved the problem for me (with Android). Thanks!
